app/controllers/api/functions.php

```diff
@@ -535,33 +574,62 @@
             }
         }
         
-        $deploymentId = $dbForProject->getId();
-        $deployment = $dbForProject->createDocument('deployments', new Document([
-            '$id' => $deploymentId,
-            '$read' => ['role:all'],
-            '$write' => ['role:all'],
-            'resourceId' => $function->getId(),
-            'resourceType' => 'functions',
-            'dateCreated' => time(),
-            'entrypoint' => $entrypoint,
-            'path' => $path,
-            'size' => $size,
-            'search' => implode(' ', [$deploymentId, $entrypoint]),
-            'activate' => $activate,
-        ]));
-
-        // Enqueue a message to start the build
-        Resque::enqueue(Event::BUILDS_QUEUE_NAME, Event::BUILDS_CLASS_NAME, [
-            'projectId' => $project->getId(),
-            'functionId' => $function->getId(),
-            'deploymentId' => $deploymentId,
-            'type' => BUILD_TYPE_DEPLOYMENT
-        ]);
+        if($chunksUploaded === $chunks) {
+            $fileSize = $deviceFunctions->getFileSize($path);
+
+            if ($deployment->isEmpty()) {
+                $deployment = $dbForProject->createDocument('deployments', new Document([
+                    '$id' => $deploymentId,
+                    '$read' => ['role:all'],
+                    '$write' => ['role:all'],
```

Comment:
```suggestion
                    '$read' => [],
                    '$write' => [],
```
The `role:all` part does not seem secure.. @TorstenDittmann  @eldadfux  should we just leave the permissions empty ?

Reply:
@christyjacob4 took from your branch?
https://github.com/appwrite/appwrite/blob/e9abe0f943b9e932bdb990fa594a9d555f5ebc4a/app/controllers/api/functions.php#L541